### PR TITLE
New package: python-watchman-1.4.1

### PR DESCRIPTION
**Pull request by sihoang**
https://pypi.org/project/pywatchman/

### CODE REVIEW

File: srcpkgs/pywatchman/template [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+short_desc="Connect and query Watchman to discover file changes (Python 2)"
+maintainer="Hoang Nguyen <hoang@wetrust.io>"
+license="BSD"
+homepage="https://facebook.github.io/watchman/"
+distfiles="${PYPI_SITE}/p/${pkgname}/${pkgname}-${version}.tar.gz"
+checksum=d0047eb275deafb0011eda0a1a815fbd9742478c3d2b5ad6956d300e447dc2f9
+alternatives="watchman-make:watchman-make:/usr/bin/watchman-make2 watchman-wait:watchman-wait:/usr/bin/watchman-wait2"
+
+post_install() {
+	vlicense PKG-INFO
```
Comment (sihoang) [inline srcpkgs/pywatchman/template]: the package from https://pypi.org/project/pywatchman/ does not have LICENSE file

File: srcpkgs/pywatchman/template [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+version=1.4.1
+revision=1
+noarch=yes
+build_style=python-module
+pycompile_module="pywatchman"
+hostmakedepends="python-devel python3-devel python-setuptools python3-setuptools"
+depends="python watchman"
+short_desc="Connect and query Watchman to discover file changes (Python 2)"
+maintainer="Hoang Nguyen <hoang@wetrust.io>"
+license="BSD"
```
Comment (sihoang) [inline srcpkgs/pywatchman/template]: same as https://pypi.org/project/pywatchman/

File: srcpkgs/pywatchman/template (hunk 1 above)
Comment (sihoang) [reply srcpkgs/pywatchman/template]: pywatch

**Comment by sihoang** [comment]
@maxice8  the original source is here https://github.com/facebook/watchman/tree/master/python
which has the LICENSE and more binaries https://github.com/facebook/watchman/tree/master/python/bin

Should I use pypi or the original? Please advise.

**Comment by sihoang** [comment]
Also, this package provides:

- A language module: `pywatchman` which should be called `python-pywatchman`. I'll change it soon.
- Multiple binaries with different names.

Should I make binaries become sub-packages? Or is it unnecessary because `alternatives` already provide the symlinks.

Your inputs are appreciated.

File: srcpkgs/pywatchman/template (hunk 2 above)
Comment (maxice8) [reply srcpkgs/pywatchman/template]: License of the project is Apache-2.0

Comment (maxice8) [reply srcpkgs/pywatchman/template]: But of the python module is BSD-3-Clause, so use this one

**Comment by maxice8** [comment]
Use the original and set python as the wrksrc

**Comment by sihoang** [comment]
Got it! I'll get it done today. Thanks for the direction.

**Comment by sihoang** [comment]
@maxice8 the issue is resolved. Please review. Thanks!

File: srcpkgs/python-pywatchman/template [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+build_style=python-module
+pycompile_module="pywatchman"
+hostmakedepends="python-devel python3-devel python-setuptools python3-setuptools"
+makedepends="python-devel python3-devel"
+depends="python watchman"
+short_desc="Connect and query Watchman to discover file changes (Python 2)"
+maintainer="Hoang Nguyen <hoang@wetrust.io>"
+license="BSD-3-Clause"
+homepage="https://github.com/facebook/watchman/tree/master/python"
+distfiles="https://github.com/facebook/watchman/archive/v4.9.0.tar.gz"
```
Comment (maxice8) [inline srcpkgs/python-pywatchman/template]: should use ${version}

File: srcpkgs/python-pywatchman/template [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+post_install() {
+	vlicense LICENSE
+}
+
+python3-pywatchman_package() {
+	short_desc="${short_desc/2/3}"
+	noarch=yes
+	pycompile_module="pywatchman"
+	hostmakedepends="python3-devel python3-setuptools"
```
Comment (maxice8) [inline srcpkgs/python-pywatchman/template]: no need to replicate hostmake and make depends here

Comment (sihoang) [reply srcpkgs/python-pywatchman/template]: ack, will fix it tonight

File: srcpkgs/python-pywatchman/template (hunk 3 above)
Comment (sihoang) [reply srcpkgs/python-pywatchman/template]: there are 2 different versions here:

The pywatchman version is 1.4.1
The pywatchman however sits in the watchman repo whose version is 4.9.0

I am thinking of declaring another variable called `watchman_version` WDYT?

File: srcpkgs/pywatchman/template (hunk 1 above)
Comment (maxice8) [reply srcpkgs/pywatchman/template]: Just put it in the distfiles then

vlicense ../LICENSE.txt

**Comment by maxice8** [comment]
Change of plans, use the PYPI tarball and fetch the license by putting it in the distfile and redirecting it into >LICENSE.txt

then use vlicense ../LICENSE.txt

File: srcpkgs/python-pywatchman/template (hunk 3 above)
Comment (maxice8) [reply srcpkgs/python-pywatchman/template]: Decided on taking the PYPI tarball approach

**Comment by sihoang** [comment]
sorry for the delay, I am working on it now

File: srcpkgs/python-pywatchman/template [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+alternatives="watchman-make:watchman-make:/usr/bin/watchman-make2 watchman-wait:watchman-wait:/usr/bin/watchman-wait2"
+
+post_install() {
+	curl https://raw.githubusercontent.com/facebook/watchman/master/python/LICENSE > LICENSE
+	vlicense LICENSE
+}
+
+python3-pywatchman_package() {
+	short_desc="${short_desc/2/3}"
+	depends="python3 watchman"
```
Comment (sihoang) [inline srcpkgs/python-pywatchman/template]: I assume we need this.

File: srcpkgs/python-pywatchman/template [hunk 6]
```diff
@@ -0,0 +1,34 @@
+# Template file for 'python-pywatchman'
+pkgname=python-pywatchman
+version=1.4.1
+revision=1
+wrksrc="pywatchman-${version}"
+build_style=python-module
+only_for_archs="x86_64 x86_64-musl"
+pycompile_module="pywatchman"
+hostmakedepends="python-devel python3-devel python-setuptools python3-setuptools"
+makedepends="curl"
```
Comment (sihoang) [inline srcpkgs/python-pywatchman/template]: to fetch the LICENSE file

**Comment by sihoang** [comment]
@maxice8 please review. 
Still pending CI.

File: srcpkgs/python-pywatchman/template (hunk 5 above)
Comment (maxice8) [reply srcpkgs/python-pywatchman/template]: yes, we need this.

File: srcpkgs/python-pywatchman/template (hunk 6 above)
Comment (maxice8) [reply srcpkgs/python-pywatchman/template]: not actually required, you can just add the path to the raw license in the distfiles= in this manner

distfiles="url-to-first-thing
 https://raw.githubusercontent.com/facebook/watchman/master/python/LICENSE>LICENSE.txt"

.txt files are merely extracted into the XBPS_BUILDDIR, you can then `vlicense ../LICENSE.txt` to install it.

File: srcpkgs/python-pywatchman/template [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+checksum=d0047eb275deafb0011eda0a1a815fbd9742478c3d2b5ad6956d300e447dc2f9
+alternatives="watchman-make:watchman-make:/usr/bin/watchman-make2 watchman-wait:watchman-wait:/usr/bin/watchman-wait2"
+
+post_install() {
+	curl https://raw.githubusercontent.com/facebook/watchman/master/python/LICENSE > LICENSE
+	vlicense LICENSE
+}
+
+python3-pywatchman_package() {
+	short_desc="${short_desc/2/3}"
```
Comment (maxice8) [inline srcpkgs/python-pywatchman/template]: It is preferable to replace Python2 and Python3 in the short_desc

File: srcpkgs/python-pywatchman/template [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+distfiles="${PYPI_SITE}/p/pywatchman/pywatchman-${version}.tar.gz"
+checksum=d0047eb275deafb0011eda0a1a815fbd9742478c3d2b5ad6956d300e447dc2f9
+alternatives="watchman-make:watchman-make:/usr/bin/watchman-make2 watchman-wait:watchman-wait:/usr/bin/watchman-wait2"
+
+post_install() {
+	curl https://raw.githubusercontent.com/facebook/watchman/master/python/LICENSE > LICENSE
+	vlicense LICENSE
+}
+
+python3-pywatchman_package() {
```
Comment (maxice8) [inline srcpkgs/python-pywatchman/template]: You need to replicate pycompile_module here as well.

File: srcpkgs/python-pywatchman/template [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+version=1.4.1
+revision=1
+wrksrc="pywatchman-${version}"
+build_style=python-module
+only_for_archs="x86_64 x86_64-musl"
+pycompile_module="pywatchman"
+hostmakedepends="python-devel python3-devel python-setuptools python3-setuptools"
+makedepends="curl"
+depends="python watchman"
+short_desc="Connect and query Watchman to discover file changes (Python 2)"
```
Comment (maxice8) [inline srcpkgs/python-pywatchman/template]: It is preferred to use `(Python2)` 

which can be replaced later on in `short_desc="${short_desc/Python2/Python3}"`

File: srcpkgs/python-pywatchman/template [hunk 10]
```diff
@@ -0,0 +1,34 @@
+# Template file for 'python-pywatchman'
+pkgname=python-pywatchman
```
Comment (maxice8) [inline srcpkgs/python-pywatchman/template]: The py prefix isn't required if you start with `python-`.

Some examples (us -> upstream):

- python-openssl -> pyOpenSSL.
- python3-xdg -> pyxdg
- python3-cups -> pycups
- python3-curl -> pycurl

File: srcpkgs/python-pywatchman/template [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+post_install() {
+	curl https://raw.githubusercontent.com/facebook/watchman/master/python/LICENSE > LICENSE
+	vlicense LICENSE
+}
+
+python3-pywatchman_package() {
+	short_desc="${short_desc/2/3}"
+	depends="python3 watchman"
+	alternatives="watchman-make:watchman-make:/usr/bin/watchman-make3 watchman-wait:watchman-wait:/usr/bin/watchman-wait3"
```
Comment (maxice8) [inline srcpkgs/python-pywatchman/template]: multiple alternatives invocations can be made on multiple lines

alternatives="
    foo:foo:usr/bin/foo
    bar:bar:/usr/bin/bar"

**Comment by sihoang** [comment]
I believe I addressed all the comments. Please review and let me know if I miss anything. CI is still pending though.

**Comment by sihoang** [comment]
oh let me squash it

File: srcpkgs/python-watchman/template [hunk 12]
```diff
@@ -0,0 +1,39 @@
+# Template file for 'python-watchman'
+pkgname=python-watchman
+version=1.4.1
+revision=1
+wrksrc="pywatchman-${version}"
+build_style=python-module
+only_for_archs="x86_64 i686 x86_64-musl"
```
Comment (maxice8) [inline srcpkgs/python-watchman/template]: no need for these

Comment (maxice8) [reply srcpkgs/python-watchman/template]: just add python3-devel and python-devel to makedepends=

**Comment by maxice8** [comment]
Please notify when proposed changes are done.

File: srcpkgs/python-watchman/template (hunk 12 above)
Comment (sihoang) [reply srcpkgs/python-watchman/template]: this is great to know!

**Pull request merged**